cdap-common/src/main/java/io/cdap/cdap/common/metadata/QueryParser.java

```diff
@@ -40,28 +43,118 @@ private QueryParser() {}
+   * By default, this method considers search items without a qualifying operator to be optional.
+   *
+   * For numeric searches, multiple comparison operators can be used.
+   * >, >=, <, <=, or = can be placed before a numeric search field to denote a
+   * greater-than, greater-than-or-equal-to, less-than, less-than-or-equal-to search, or equality search, respectively.
+   * Search items without a comparison operator are considered string-based searches.
    *
    * @param query the raw search string
    * @return a list of QueryTerms
    */
   public static List<QueryTerm> parse(String query) {
     List<QueryTerm> queryTerms = new ArrayList<>();
-    for (String term : Splitter.on(SPACE_SEPARATOR_PATTERN)
+    for (String inputTerm : Splitter.on(SPACE_SEPARATOR_PATTERN)
         .omitEmptyStrings().trimResults().split(query)) {
-      queryTerms.add(parseQueryTerm(term));
+      queryTerms.add(parseQueryTerm(inputTerm));
     }
     return queryTerms;
   }
 
+  /**
+   * Extracts the raw value of the input term, given that terms can follow a key:[comparison-operator]value syntax.
+   * This method removes any syntactic characters from the input string, including comparison and wildcard operators.
+   * As an example, extractTermValue("key:>=30") returns "30".
+   *
+   * Note that this method removes comparison operators from alphabetic strings as well, even though they do not qualify
+   * for numeric search.
+   * As an example, extractTermValue("+>=thirty") returns "thirty".
+   *
+   * @param term the search term, with all syntactic operators included
+   * @return the raw value of the search term, with all syntactic operators excluded
+   */
+  public static String extractTermValue(String term) {
+    term = lastSubTerm(term);
+    if (term.startsWith(">") || term.startsWith("<")) {
+      term = term.substring(1);
+    }
+    if (term.startsWith("=")) {
+      term = term.substring(1);
+    }
+    if (term.endsWith("*")) {
+      term = term.substring(0, term.length() - 1);
+    }
+
+    return term;
+  }
+
   private static QueryTerm parseQueryTerm(String term) {
-    if (term.charAt(0) == REQUIRED_OPERATOR && term.length() > 1) {
-      return new QueryTerm(term.substring(1), Qualifier.REQUIRED);
+    SearchType parsedSearchType = isValidNumber(term)
+        ? SearchType.NUMERIC : SearchType.STRING;
+
+    Qualifier parsedQualifier = (term.startsWith(REQUIRED_OPERATOR)
+        && term.length() > 1) || parsedSearchType == SearchType.NUMERIC
+        ? Qualifier.REQUIRED : Qualifier.OPTIONAL;
+
+    String parsedTerm = term.startsWith(REQUIRED_OPERATOR)
+        && term.length() > 1
+        ? term.substring(1) : term;
+
+    Comparison parsedComparison = parsedSearchType == SearchType.STRING
+        ? Comparison.EQUALS : findComparison(term);
+
+    return new QueryTerm(parsedTerm, parsedQualifier, parsedSearchType, parsedComparison);
+  }
+
+  private static boolean isValidNumber(String term) {
```

Comment:
I think it would be better to get the search type from here.

Reply:
Addressed in the next commit.